Update paper.md (#2)
yaml2sbml-dev/paper: paper.md

# Summary

> Ordinary differential equation (ODE)  models are used in various fields of natural sciences to describe dynamic processes. In the life sciences, they are mostly stored and exchanged using the the Systems Biological Markup Language (SBML),  a widely adopted community standard based on XML. The Parameter Estimation table (PEtab) format builds on SBML to describe parameter estimation problems. There exists a large number of software pipelines for simulating SBML models and performing parameter estimation for PEtab problems Hence, specifying ODE models in SBML and parameter estimation problem in PEtab provides access to a broad spectrum of tools. However, SBML is considered to be neither human-readable nor human-writable. An easy-to-use approach to construct the SBML/PEtab models for a given ODE model will facilitate model generation. 

Reviewer: ```suggestion
Ordinary differential equation (ODE)  models are used in various fields of natural sciences to describe dynamic processes. In the life sciences, they are mostly stored and exchanged using the the Systems Biological Markup Language (SBML),  a widely adopted community standard based on XML. The Parameter Estimation table (PEtab) format builds on SBML to describe parameter estimation problems. There exists a large number of software pipelines for simulating SBML models and performing parameter estimation for PEtab problems. Hence, specifying ODE models in SBML and parameter estimation problem in PEtab provides access to a broad spectrum of tools. However, SBML is considered to be neither human-readable nor human-writable. An easy-to-use approach to construct the SBML/PEtab models for a given ODE model will facilitate model generation. 
```

> In this contribution, we present `yaml2sbml`, a Python tool for converting ODE models specified in an easy-to-read and -write YAML file into SBML/PEtab.  `yaml2sbml` comes with a format validator for the input YAML, a command-line interface and a model editor to: 1) create an ODE model programmatically in Python which can then be written into SBML/PEtab or YAML and 2) edit an ODE model previously encoded in YAML. Several examples illustrate the use of `yaml2sbml` on realistic problems.

Reviewer: ```suggestion
In this contribution, we present `yaml2sbml`, a Python tool for converting ODE models specified in an easy to read and write YAML file into SBML/PEtab.  `yaml2sbml` comes with a format validator for the input YAML, a command-line interface and a model editor to: 1) create an ODE model programmatically in Python which can then be written into SBML/PEtab or YAML and 2) edit an ODE model previously encoded in YAML. Several examples illustrate the use of `yaml2sbml` on realistic problems.
```

Reviewer: Since it is `easy to read` in all other parts of the draft...

# Statement of need

Ordinary differential equations (ODEs) are a vital tool in mechanistic modelling of biological processes. ODE models span a wide range of scales and applications, from molecular to population level [@FroehlichKes2018, @RaimundezDud2020_arxiv] and arise e.g. as discretizations of Partial Differential Equation models [@FischerFie2019] or moments of stochastic systems [@Engblom2006].

The Systems Biological Markup Language (SBML) is a widely adopted community standard for specifying biological reaction networks [@HuckaFin2003]. [sbml.org](http://sbml.org/SBML_Software_Guide/SBML_Software_Summary#cat_12) lists more than 100 software tools that accept SBML as their input format for dynamic model simulation, among them [@HoopsSah2006; @FroehlichWei2020; @RaueSte2015; @KaschekMad2019]. The [biomodels.net](biomodels.net) data base contains more than 1,000 published models in the SBML format [@NovereBor2006].

> Model parameters can be estimated from data by formulating a likelihood function. To do so, system states must be mapped to measured quantities by observable functions, and a measurement noise model must be specified. The PEtab format was recently introduced to complement SBML by tab-separated value files specifying observables, measurements, experimental conditions, and estimated parameters [@SchmiesterSch2020]. Currently 9 software toolboxes support PEtab as an input format, among them COPASI, D2D and AMICI/pyPESTO.

Reviewer: ```suggestion
Model parameters can be estimated from data by formulating a likelihood function. Therefore, the  system states must be mapped to measured quantities by observable functions, and a measurement noise model must be specified. The PEtab format was recently introduced to complement SBML by tab-separated value files specifying observables, measurements, experimental conditions, and estimated parameters [@SchmiesterSch2020]. Currently 9 software toolboxes support PEtab as an input format, among them COPASI, D2D and AMICI/pyPESTO.
```

What about posterior definition?

> Despite its broad usage, constructing an SBML model from scratch is often tedious, as SBML is considered neither human-readable nor -writeable. Therefore, various approaches to facilitate model construction from text-based input formats or in code have been presented [@BornsteinKea2008; @CannistraMed2015; @GomezHuc2016; @SmithBer2009; @Poolman2006]. The tool by [@GomezHuc2016] offers a text-based input format tightly tied to MATLAB, as the tool translates MATLAB code into SBML. Other tools have a text-based input format centered around chemical reactions and not around ODEs directly [e.g. @Poolman2006], or only offer text-based [@SmithBer2009], or only programmatic Python-based access [@CannistraMed2015], but not both interchangeably. Also, none of the aforementioned tools support PEtab or other formats for specifying parameter estimation problems.

Reviewer: sentence was introduced by Jan and makes sense, since Non-SBML persons are also a traget group of our tool?

Reviewer: ```suggestion
Despite its broad usage, constructing an SBML model from scratch is often tedious, as SBML is considered neither human-readable nor -writeable. In particular for persons from other research fields who might be interested in using the broad spectrum of available tools, it is demanding. Therefore, various approaches to facilitate model construction from text-based input formats or in code have been presented [@BornsteinKea2008; @CannistraMed2015; @GomezHuc2016; @SmithBer2009; @Poolman2006]. The tool by [@GomezHuc2016] offers a text based input format, that is tightly tied to MATLAB, as the tool translates MATLAB code into SBML. Other tools have a text based input format that is centered around chemical reactions and not around ODEs directly [e.g. @Poolman2006], or only offer a text-based [@SmithBer2009] or only a Python-based way of defining SBML models [@CannistraMed2015], but not both at the same time interchangeably. Also, none of the aforementioned tools support PEtab or other formats for specifying parameter estimation problems.
```
